openmc/deplete/microxs.py

```diff
@@ -66,29 +68,40 @@ def from_model(cls,
             Cross section data in [b]
 
         """
-        # Set up the reaction tallies
+        # Save any original tallies on the model
         original_tallies = model.tallies
-        original_materials = deepcopy(model.materials)
-        xs = {}
-        reactions, burnable_nucs, diluted_materials = cls._add_dilute_nuclides(
-            chain_file, model, dilute_initial)
-        model.materials = diluted_materials
 
+        # Determine what reactions and nuclides are available in chain
+        if chain_file is None:
+            chain_file = openmc.config.get('chain_file')
+            if chain_file is None:
+                raise DataError(
+                    "No depletion chain specified and could not find depletion "
+                    "chain in openmc.config['chain_file']"
+                )
+        chain = Chain.from_xml(chain_file)
+        if reactions is None:
+            reactions = chain.reactions
+        if not nuclides:
+            cross_sections = _find_cross_sections(model)
+            nuclides_with_data = _get_nuclides_with_data(cross_sections)
+            nuclides = [nuc.name for nuc in chain.nuclides
+                        if nuc.name in nuclides_with_data]
+
+        # Set up the reaction rate and flux tallies
         energy_filter = openmc.EnergyFilter(energy_bounds)
-        if isinstance(reaction_domain, openmc.Material):
-            domain_filter = openmc.MaterialFilter([reaction_domain])
-        elif isinstance(reaction_domain, openmc.Cell):
-            domain_filter = openmc.CellFilter([reaction_domain])
-        elif isinstance(reaction_domain, openmc.Universe):
-            domain_filter = openmc.UniverseFilter([reaction_domain])
+        if isinstance(domain, openmc.Material):
+            domain_filter = openmc.MaterialFilter([domain])
+        elif isinstance(domain, openmc.Cell):
+            domain_filter = openmc.CellFilter([domain])
+        elif isinstance(domain, openmc.Universe):
+            domain_filter = openmc.UniverseFilter([domain])
         else:
-            raise ValueError(f"Unsupported domain type: {type(reaction_domain)}")
+            raise ValueError(f"Unsupported domain type: {type(domain)}")
 
-        # TODO: Right now, we use all nuclides from the material but it probably
-        # should be based on the burnable nuclides
         rr_tally = openmc.Tally(name='MicroXS RR')
         rr_tally.filters = [domain_filter, energy_filter]
-        rr_tally.nuclides = reaction_domain.get_nuclides()
+        rr_tally.nuclides = nuclides
```

Comment:
Good change. We'll def want ALL nuclides that could show up in a domain, not just ones that are currently there.